tests/instantiation_test.py

```diff
@@ -7,10 +7,9 @@
 from pykern.pkdebug import pkdp, pkdlog
 from pykern.pkcollections import PKDict
 import pytest
-from rslaser.rspulse.pulse import LaserPulse, LaserPulseSlice, InvalidLaserPulseInputError
+from rslaser.rspulse.pulse import LaserPulse, LaserPulseSlice
```

Comment:
Import just the module and get the classes off of the module.

Reply:
there would be a lot of pulse.LaserPulse() and pulse.LaserPulseSlice(). Should I still do it that way?

Reply:
Yes, always import a module and get the object from it.

The only time it is up to preference is doing
```python
import rslaser.rspulse.pulse
x = rslaser.rspulse.pulse.LaserPulse()
# versus
from rslaser.rspulse import pulse
x = pulse.LaserPulse()
```

The first is preferred because it has more context. But, it can be cumbersome with long names used all over the place.